code/modules/clothing/gloves/miscellaneous.dm

```diff
@@ -64,13 +64,21 @@
 	transfer_prints = TRUE
 	var/warcry = "AT"
 
-/obj/item/clothing/gloves/rapid/Touch(mob/living/target,proximity = TRUE)
+/obj/item/clothing/gloves/rapid/Touch(atom/A, proximity)
 	var/mob/living/M = loc
-
-	if(M.a_intent == INTENT_HARM)
-		M.changeNext_move(CLICK_CD_RAPID)
-		if(warcry)
-			M.say("[warcry]", ignore_spam = TRUE, forced = "north star warcry")
+	if(A in range(1, M))
+		if(isliving(A) && M.a_intent == INTENT_HARM)
+			M.changeNext_move(CLICK_CD_RAPID)
+			if(warcry)
+				M.say("[warcry]", ignore_spam = TRUE, forced = "north star warcry")

```

Comment:
will this hit people? theres no attack_hand

Reply:
it doesn't need one, this is just the check that checks if you can hit a guy before saying the warcry. Attack_hand is already called here. The reason the next proc uses attack_hand is cuz you arent clicking the guy directly